New Job Opportunity also reflect on ColoredCow Website

am-Anmol opened the pull request:
Targets #794 

### Description
**Job opportunities will `create/update/delete` on the `ColoredCow website` also if updated from `Employee Portal`.**
- Changes made in `Modules/HR/Observers/Recruitment/JobObserver.php` :
     - **`Created` function (working):**
          - Check if `status` is `published` or not, if it is `not` then do nothing.
          - If it is `published` then creating a new entry of data in `wp_post` table with `post_title,post_content,post_type` and a new entry in `wp_postmeta` table with `hr_id` and value of `hr_id` from employee portal `hr_jobs` table.
     - **`Updated` function (working):**
          - Check if `status` is `published` or not, if it is `not` then do nothing.
          - If it is `published` then creating a new entry of data in `wp_post` table with `post_title,post_content,post_type` and a new entry in `wp_postmeta` table with `hr_id` and value of `hr_id` from employee portal `hr_jobs` table.
          - Then finding `post id` by matching `hr_id` in `wp_postmeta` table (Website) with `hr_id` in `hr_jobs` table (employee portal).
          - After that finding an old post with `post id` and deleting the old post.
     - **`Deleted` function (working):**
          - Matching `post_title` and `post_type` with records and deleting the post.
      
- **Prerequisites Required:**
     - Update `config/database.php` with the correct database configurations.
     - ![image](https://user-images.githubusercontent.com/48876126/129408078-df379005-bfcb-48cc-87af-37fb25d625f6.png)

**References:**
Check out the Corcel package to easily create entries in the WordPress database from Laravel
https://github.com/corcel/corcel
### Checklist:
<!--- Mark the checkboxes accordingly. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I have performed a self-review of my own code.

GauravGusain98 commented on Modules/HR/Observers/Recruitment/JobObserver.php:
```diff
@@ -16,5 +17,49 @@ class JobObserver
+            $Corcel = new Corcel();
+            $Corcel->post_title = $data['title'];
+            $Corcel->post_content = $data['description'];
+            $Corcel->post_type = 'career';
```
Please use the value from config files instead of hardcoding. This way if we want to change the value of post_type, we have to only change that in one place instead of changing at every place.

GauravGusain98 commented on Modules/HR/Observers/Recruitment/JobObserver.php:
```diff
@@ -16,5 +17,49 @@ class JobObserver
+            $Corcel = new Corcel();
+            $Corcel->post_title = $data['title'];
+            $Corcel->post_content = $data['description'];
+            $Corcel->post_type = 'career';
```
Please use the value from config files instead of hardcoding. This way if we want to change the value of post_type, we have to only change that in one place instead of changing at every place.

am-Anmol replied:
Ok, I make these changes.

GauravGusain98 commented on Modules/HR/Observers/Recruitment/JobObserver.php:
```diff
@@ -16,5 +17,49 @@ class JobObserver
+        $data = request()->all();
+
+        if ($data['status'] == 'published') {
+            $Corcel = new Corcel();
+            $Corcel->post_title = $data['title'];
+            $Corcel->post_content = $data['description'];
+            $Corcel->post_type = 'career';
+            $Corcel->save();
+            $Corcel->saveMeta('hr_id', $job['id']);
+            $post = Corcel::hasMeta('hr_id', $job['id'])->first();
+            $corcel = Corcel::find($post->ID);
+            $corcel->delete();
```
This is not looking correct to me as we need to update the record here. But in this case, we are creating a new entry and deleting the old one. Please use the update approach for this.

am-Anmol replied:
OK, I'll update this approach.

GauravGusain98 reviewed (changes requested):
Please make the suggested changes.

am-Anmol commented:
**Update function approach:**
- Check if `status` is `published` or not, if it is `not` then do nothing.
- If it is `published` then find `post ID` in `wp_post` table using `job_id`.
- Then finding `post` with `post ID` and updating respective columns.

GauravGusain98 commented on Modules/HR/Observers/Recruitment/JobObserver.php:
```diff
@@ -16,5 +17,42 @@ class JobObserver
+    {
+        $data = request()->all();
+
+        if ($data['status'] == 'published') {
```
Just a doubt, can there be a case when the status is not published and we can update the job? If yes, then we need to update the job details for both the case.

am-Anmol replied:
Yes, this can be a possible scenario then we have to update both of them.
If we update the job from the portal and mark status to published it also gets updated on a website post.

GauravGusain98 replied:
Let me rephrase my question, 
I was saying that suppose a job is not published but you updated the job details in the portal. In this case as well, the job details should get updated in the website post. But as per the code, it looks like that the job will only get updated when the status is published.

Also what about the case when we update the job from `published` to `draft`. I don't think in that case it will mark the job as a draft on the website. 

Let me know if there is any doubt in understanding.

GauravGusain98 reviewed (commented):
Please check the comment

GauravGusain98 commented:
Please check the above comment

GauravGusain98 reviewed (changes requested):
Please check the comments and if required do the requested changes.

GauravGusain98 commented on Modules/HR/Observers/Recruitment/JobObserver.php:
```diff
@@ -16,5 +17,42 @@ class JobObserver
         $job->rounds()->attach(Round::all()->pluck('id')->toArray());
+        $data = request()->all();
+
+        if ($data['status'] == 'published') {
+            $Corcel = new Corcel();
+            $Corcel->post_title = $data['title'];
+            $Corcel->post_content = $data['description'];
+            $Corcel->post_type = config('hr.post-type.career');
+            $Corcel->save();
+            $Corcel->saveMeta('hr_id', $job['id']);
+        }
+    }
```
Let me rephrase my question, 
I was saying that suppose a job is not published but you updated the job details in the portal. In this case as well, the job details should get updated in the website post. But as per the code, it looks like that the job will only get updated when the status is published.

Also what about the case when we update the job from `published` to `draft`. I don't think in that case it will mark the job as a draft on the website. 

Let me know if there is any doubt in understanding.

am-Anmol replied:
Yeah, I have understood your scenario and updated the code as per all scenarios possible while creating and updating any post through an employee portal.

GauravGusain98 reviewed (changes requested):
Please do the following changes.

am-Anmol commented:
**Changes in created and updated function:**
- Created:
    - Checking if the status is `not published` then saving the post as a draft.
- Updated:
   - Checking if the status is `published` then updating the post as a `publish`, otherwise updating as `draft` only.

GauravGusain98 commented on Modules/HR/Observers/Recruitment/JobObserver.php:
```diff
@@ -16,5 +17,50 @@ class JobObserver
+        $Corcel->post_title = $data['title'];
+        $Corcel->post_content = $data['description'];
+        $Corcel->post_type = config('hr.post-type.career');
+        if ($data['status'] == 'published') {
+            $Corcel->post_status = 'publish';
+        } else {
+            $Corcel->post_status = 'draft';
+        }
```
```suggestion
      $Corcel->post_status = $data['status'] == 'published' ? 'published' : 'draft';
```

am-Anmol replied:
Ok I make changes

GauravGusain98 reviewed (changes requested):
Please check the suggestion

GauravGusain98 reviewed (approved):
Looks good to me
